application-mocca-calendar-api/src/main/java/org/xwiki/contrib/moccacalendar/internal/importJob/CalendarEventImporter.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.moccacalendar.internal.importJob;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Safelist;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.contrib.moccacalendar.internal.EventConstants;
+import org.xwiki.fullcalendar.model.CalendarEvent;
+import org.xwiki.fullcalendar.model.RecurrentEventModification;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.SpaceReference;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.stability.Unstable;
+import org.xwiki.wysiwyg.converter.HTMLConverter;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Helper class for processing an imported event.
+ *
+ * @version $Id$
+ * @since 2.14
+ */
+@Unstable
+@Component(roles = CalendarEventImporter.class)
+@Singleton
+public class CalendarEventImporter
+{
+    private static final String NEW_EVENT_HOME = "WebHome";
+
+    @Inject
+    private HTMLConverter htmlConverter;
+
+    @Inject
+    private Provider<XWikiContext> wikiContextProvider;
+
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> documentReferenceResolver;
+
+    /**
+     * Creates the class objects specific to a Mocca calendar event and adds them to the given {@link XWikiDocument}.
+     *
+     * @param eventDoc the {@link XWikiDocument} where the objects will be added.
+     * @param component contains the data used to populate the objects.
+     * @throws XWikiException if there are any issues when creating the {@link BaseObject} for the given document.
+     */
+    public void createCalendarObjects(XWikiDocument eventDoc, CalendarEvent component) throws XWikiException
```

Comment:
Maybe `importCalendarEvent`? For me this is what you do. Only the save is done outside of this method, otherwise this is the place where the import in the wiki is done.
I think it will feel more coherent, to not simply have a helper class with some methods